hpke.go

```diff
@@ -184,21 +184,38 @@ func keySchedule(suite CipherSuite, mode Mode, sharedSecret, info, psk, pskID []
 	return params, nil
 }
 
+// Context represents an HPKE context encoded on the wire.
+type Context struct {
+	Role           Role
+	KdfId, AeadId  uint16
```

Comment:
We also need to store the KEM ID, as that's used when constructing the ciphersuite ID, and that's used by the `Export` API. Can we add a test that (1) exports a secret, (@) marshals and then unmarshals the context, and (3) exports the same secret and compares it to the output from (1)?

Reply:
Done. Good catch!